src/type/classes/StringType.ts

```diff
@@ -69,6 +91,19 @@ export class StringType extends AbstractType<schema.StringSchema> {
         ctx.js(/* js */ `if(${r}.length > ${max}) return ${err};`);
       }
     }
+    
+    // Handle format validation
+    if (format) {
+      const formatErr = ctx.err(ValidationError.STR, path);
+      const validateFn = ctx.codegen.linkDependency(validateStringFormat);
+      ctx.js(/* js */ `if(!${validateFn}(${r}, "${format}")) return ${formatErr};`);
+    } else if (ascii) {
+      // Backward compatibility: use ASCII validation if ascii=true and no format specified
+      const asciiErr = ctx.err(ValidationError.STR, path);
+      const validateFn = ctx.codegen.linkDependency(validateStringFormat);
```

Comment:
Link `isAscii` here and call it:

```suggestion
      const validateFn = ctx.codegen.linkDependency(isAscii);
```